modules/samples/pages/dex.adoc

```diff
@@ -1,3 +1,96 @@
+# DEX サンプル
+
+IC 上で DEFI アプリケーションを実現するためには、DEFI の Canister は token Canister や ledger Canister と相互に対話する必要があります。このサンプル Dapp ではこれらの相互対話を用意にする方法を説明します。
+
+サンプルのDEFIは https://github.com/dfinity/examples/tree/master/motoko/defi[Motoko] と https://github.com/dfinity/examples/tree/master/rust/defi[Rust] で実装されており、 https://gzz56-daaaa-aaaal-qai2a-cai.ic0.app/[ IC 上のここで動作中] を見ることが出来ます。
+
+## アーキテクチャ
+
+IC の設計ではより複雑なオンチェーン演算処理が可能です。汎用ストレージと組み合わせれば、オンチェーンでのオーダーブックも可能です。このサンプルコードではこれらの機能を利用して、ユーザーの残高と注文を取引所 Canister に保存します。サンプルの取引所機能は以下のステップに要約することが出来ます。
+
+* 取引所が資金の保管（ ICP と他のトークンでは仕組みが異なる、後述）。
+* 取引所が内部バランスブックを更新する。
+* ユーザーは取引所で取引を行い、取引所は内部バランスブックを更新する。
+* 取引所から資金を引き出すと、保管されていた資金はユーザーに戻される。
+
+### インターフェイス
+
+取引所からユーザー固有の元帳アカウント識別子をリクエストします。この一意のアカウント識別子は、取引所の元帳アカウントのユーザー固有のサブアカウントを表し、ユーザーのデポジットを区別できるようにします。
+
+```candid
+getDepositAddress: () -> (blob);
+```
+
+取引所へのユーザーのデポジットを開始します。ユーザーが ICP をデポジットしたい場合、取引所はユーザー固有のデポジットアドレスからそのデフォルトのサブアドレスに資金を移動し、DEX でユーザーの ICP 残高を調整します。ユーザーが DIP トークンのデポジットを希望している場合、取引所は承認された資金をそのトークン口座に移動しようとし、ユーザーの残高を調整します。
+
+```candid
+deposit: (Token) -> (DepositReceipt);
+```
+
+取引所へ出金依頼をします。取引所はユーザーが十分な残高を持っている場合、ユーザーに資金を送り返します。
+
+```candid
+withdraw: (Token, nat, principal) -> (WithdrawReceipt);
+```
+
+取引所に新しい注文を出します。注文が適正であった場合、注文は執行されます。
+
+```candid
+placeOrder: (Token, nat, Token, nat) -> (OrderPlacementReceipt);
+```
+
+ユーザーに発注をキャンセル出来るようにします。
+
+```candid
+cancelOrder: (OrderId) -> (CancelOrderReceipt);
+```
+
+ユーザーの持っている取引所にある特定トークン残高を要求します。
+
+```candid
+getBalance: (Token) -> (nat) query;
+```
+
+### 手数料
+
+取引から手数料を差し引くのは取引所の責任です。取引所が出金等の内部送金の手数料を支払わなければならないため重要です。
+
+## トークン取引所の具体的な検証
+
+このセクションには、取引所の中核機能の詳細な検証が含まれています。ほとんどの相互対話には複数のステップを要求され、提供されるフロントエンドを使用することで簡略化されます。取引所 Canister の機能は公開されているので、スキルのあるユーザーは ``dfx`` を使用して取引所とやり取りすることができます。
+
+### ICP のデポジット
+
+ledger Canister は独自のインターフェースを提供するため、ICP の対話において別途解決しなければならない諸問題があります。
```

Comment:
so that 構文では、so that 以前に示したアクションの理由をso that の後ろに示しています。「別々に解決しなければならない処理」が下の箇条書きです
```suggestion
ICP のやり取りにおいて別々に解決（Resolve）しなければならない処理があるため、Ledger Canister は独自のインターフェースを提供しています。
```